js/gateio.js

```diff
@@ -2522,6 +2522,140 @@ module.exports = class gateio extends Exchange {
         return response;
     }
 
+    parsePosition (position, market = undefined) {
+        //
+        //     {
+        //         value: "12.475572",
+        //         leverage: "0",
+        //         mode: "single",
+        //         realised_point: "0",
+        //         contract: "BTC_USDT",
+        //         entry_price: "62422.6",
+        //         mark_price: "62377.86",
+        //         history_point: "0",
+        //         realised_pnl: "-0.00624226",
+        //         close_order:  null,
+        //         size: "2",
+        //         cross_leverage_limit: "25",
+        //         pending_orders: "0",
+        //         adl_ranking: "5",
+        //         maintenance_rate: "0.005",
+        //         unrealised_pnl: "-0.008948",
+        //         user: "663337",
+        //         leverage_max: "100",
+        //         history_pnl: "14.98868396636",
+        //         risk_limit: "1000000",
+        //         margin: "0.740721495056",
+        //         last_close_pnl: "-0.041996015",
+        //         liq_price: "59058.58"
+        //     }
+        //
+        const contract = this.safeValue (position, 'contract');
+        market = this.safeMarket (contract, market);
+        const now = this.milliseconds ();
+        const size = this.safeNumber (position, 'size');
+        let side = undefined;
+        if (size > 0) {
+            side = 'buy';
+        } else if (size < 0) {
+            side = 'sell';
+        }
+        const maintenanceRate = this.safeValue (position, 'maintenance_rate');
+        const markPrice = this.safeValue (position, 'mark_price');
+        // const contractSize = this.safeValue (position, 'size');
+        const notional = this.safeValue (position, 'value');
+        const collateral = this.safeValue (position, 'margin');
+        const marginRatio = Precise.stringDiv (maintenanceRate, collateral);
+        const initialMargin = '';
+        const unrealisedPnl = this.safeValue (position, 'unrealised_pnl');
+        return {
+            'info': position,
+            'symbol': market['symbol'],
+            'timestamp': now,
+            'datetime': this.iso8601 (now),
+            'initialMargin': undefined,
+            'initialMarginPercentage': Precise.stringDiv (initialMargin, notional),
+            'maintenanceMargin': Precise.stringMul (maintenanceRate, notional),
```

Comment:
you need to do `this.parseNumber (Precise.stringMul`

since stringMul returns a string and we want to have a number here